webapp/graphite/render/views.py

```diff
@@ -48,6 +50,36 @@
 from django.utils.cache import add_never_cache_headers, patch_response_headers
 
 
+def prefetchRemoteData(requestContext, targets):
+  """Prefetch a bunch of path expressions and stores them in the context.
+
+  The idea is that this will allow more batching that doing a query
+  each time evaluateTarget() needs to fetch a path. All the prefetched
+  data is stored in the requestContext, to be accessed later by datalib.
+  """
+  log.rendering("Prefetching remote data")
+  pathExpressions = extractPathExpressions(targets)
+  results = STORE.prefetch(requestContext, pathExpressions)
+
+  prefetched = collections.defaultdict(list)
+  for result in results:
+    fetched = wait_for_result(result)
```

Comment:
Right, I assumed this didn't really matter because we rarely mix local and remote together, but I'll make sure we do that as late as we can. New patch coming on monday :)